### DISCUSSION TITLE
elegant solution to dynamicly adjust number of obstacle points?

### DISCUSSION THREAD

**Discussion by lxbnm** [casadi/casadi#3364] [Q&A]
Hi i just started using casadi and i am currently implementing an MPC for trajectory planing and control of a small race car. So far my MPC does a good job following a reference path and doing it using Opti() was very intuitive so thanks for that! 

Next step would be avoiding obstacles. Most common technique is modeling the race car and the obstacle as circles. I use a hard constraint to enforce that the center points of the circles have to be a fixed distance away from each other.This is combined with a cost that increases when the circles come close to each other. So far so good but here comes my question:

Is there an elegant way to handle a dynamic number of static obstacles in casadi? Most of the work i looked into just hard code a few obstacles in the simulation. In the real world I first have to "see" a obstacle before I can react. And usually we don't know beforehand if we will have no, one, two or maybe even five obstacles at the same time. So how can I deal with that? I was thinking about something like:

opti.subject_to(casadi.sqrt((x-x_obs)**2+(y-y_obst)**2) >= d_min)  #with obs = 1,...,M

Which means I need a way to tell casadi how big M is so that casadi can perform the constraint for each point whether it is zero or five

**Comment by paulmteleo** [comment]
In casadi, I would create a parameter vector that is big enough to encode the the parameters for the maximum number of obstacles, lets say this max is 10.   In the control loop I would dynamically adjust the parameters for active obstacles -- up to 10.     If (for example) there are only 3 obstacles, I would code those 3 into the parameter vector, and I would code the other 7 to be far away from the robot so they are inactive constraints in the optimization.  I am not sure how to do this in opti.
